src/test/ui/suggestions/type-ascription-instead-of-path-2.stderr

```diff
@@ -0,0 +1,13 @@
+error: expected `::`, found `(`
+  --> $DIR/type-ascription-instead-of-path-2.rs:2:55
+   |
+LL |     vec![Ok(2)].into_iter().collect:<Result<Vec<_>,_>>()?;
+   |                                    -                  ^ expected `::`
```

Comment:
The "expected" bit seems to be in the wrong place?

Reply:
The expected is coming from it trying to parse `<Result<Vec<_>,_>>::AssocItem`. That expected `::` is coincidental to the actual error, `.collect:` was meant to be `.collect::`.

Reply:
Yes sure; that seems right. However, the diagnostic also seems confusing this way?

r=me with the other comment fixed if yo don't think so.

Reply:
Made a change in one of the related methods so that the output is now a suggestion for `::` in the right place, which hopefully is less confusing.